iterator.go

```diff
@@ -0,0 +1,77 @@
+//go:build goexperiment.rangefunc
+
+package lo
+
+// Iterator TODO
+type Iterator[V any] func(func(int, V) bool)
+
+// ToIterator TODO
+func ToIterator[V any](collection ...V) Iterator[V] {
```

Comment:
`ToIterator` might already be implemented in the go standard library as `slices.All`. 

As per this comment: https://github.com/gomoni/it/issues/12#issuecomment-2234202444